ci/docker/Dockerfile

```diff
@@ -1,13 +1,26 @@
-FROM centos:6.10
+FROM centos:8 as base
 
 # CentOS 6 packages are no longer hosted on the main repository, instead they are in the CentOS Vault
-RUN sed -i 's/^mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Base.repo && \
-    sed -i 's/#baseurl=http:\/\/mirror.centos.org\/centos\/$releasever/baseurl=http:\/\/vault.centos.org\/6.10/g' /etc/yum.repos.d/CentOS-Base.repo
+# RUN sed -i 's/^mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Base.repo && \
+#     sed -i 's/#baseurl=http:\/\/mirror.centos.org\/centos\/$releasever/baseurl=http:\/\/vault.centos.org\/6.10/g' /etc/yum.repos.d/CentOS-Base.repo
 
 # Set up additional build tools
 RUN yum -y update && yum clean all
-RUN yum -y install gcc curl openssl openssl-devel ca-certificates tar perl perl-Module-Load-Conditional && yum clean all
+RUN yum -y install gcc git curl openssl openssl-devel ca-certificates tar perl perl-Module-Load-Conditional && yum clean all
 
 # Install Rust
 RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable
 ENV PATH="/root/.cargo/bin:${PATH}"
+
+FROM base as openssl
+
+ENV CARGO_HOME=/root/workspace/.cargo
```

Comment:
We set this here so cargo operates from the mapped volume on the host. That's the magic that lets us cache the `target` directory in that GHA cache action.